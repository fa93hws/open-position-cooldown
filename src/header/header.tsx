import * as React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Typography from '@material-ui/core/Typography';
import Link from '@material-ui/core/Link';
import { Home, Info } from '@material-ui/icons';
import { makeStyles, withTheme, WithTheme } from '@material-ui/core/styles';
import { Link as ReactRouterLink } from 'react-router-dom';

import { pagePath } from '../routes';
import { sizes } from '../styles/styles';

const useHeaderStyles = makeStyles((theme) => ({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
    height: sizes[3],
  },
  iconWrapper: {
    color: theme.palette.primary.contrastText,
  },
}));

export const Header = withTheme((props: WithTheme) => {
  const styles = useHeaderStyles(props.theme);
  return (
    <AppBar position="static" className={styles.container}>
      <Link
        className={styles.iconWrapper}
        component={ReactRouterLink}
        to={pagePath.home}
      >
        <Home fontSize="large" />
      </Link>
      <Typography variant="h4" component="h1">
        买股冷静期
      </Typography>
      <Link
        className={styles.iconWrapper}
        component={ReactRouterLink}
        to={pagePath.about}
      >
        <Info fontSize="large" />
      </Link>
    </AppBar>
  );
});
