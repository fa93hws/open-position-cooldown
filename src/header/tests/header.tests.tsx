import * as React from 'react';

import { Header } from '../header';

describe('Header', () => {
  it('renders this', () => {
    expect(<Header />).toMatchRenderedSnapshot();
  });
});
