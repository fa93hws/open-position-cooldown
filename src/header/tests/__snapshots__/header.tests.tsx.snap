// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders this 1`] = `
<header
  class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-container-1 MuiPaper-elevation4"
>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-iconWrapper-2 MuiTypography-colorPrimary"
    href="/"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
      />
    </svg>
  </a>
  <h1
    class="MuiTypography-root MuiTypography-h4"
  >
    买股冷静期
  </h1>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-iconWrapper-2 MuiTypography-colorPrimary"
    href="/about"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
      />
    </svg>
  </a>
</header>
`;
